test/parallel/test-fs-mkdtemp.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+common.refreshTmpDir();
+
+fs.mkdtempSync(common.tmpDir + '/foo.XXXXXX');
+assert(fs.readdirSync(common.tmpDir).some(function(folder) {
+    return folder.indexOf('foo.') === 0 && folder.length === 'foo.XXXXXX'.length;
+}));
+
+fs.mkdtemp(common.tmpDir + '/bar.XXXXXX', function(err, folder) {
```

Comment:
You can use `common.mustCall(function(...`, so that it will fail the test if the callback is not invoked. Since we have the `assert` in the callback, it would be better to make sure that the callback got fired at least once.